### DISCUSSION TITLE
Center the graph when expand a node

### DISCUSSION THREAD

**Discussion by saomartinho** [xyflow/xyflow#2128] [Q&A]
# Question
The structure I am designing, there are nodes that have children and there is an action to expand / collapse, which will display the children, depending on the action.

The problem I'm encountering is that when I expand I'm not able to center the graphic with `fitView` and I can't understand why. Can see the video.

But when I make the fitView through the controls, it works.

Basically, the code I have is as follows:

```js
const toggleExpand = () => {
   ... // update the nodes and the edges.

   reactFlowInstance.fitView();
};
```
Any suggestion?
Thanks in advance for your help!

# Your Example Website or App
No response

# Steps to Reproduce the Bug or Issue
/

# Expected behavior
1 - Expand a node
2 - Center the graph on center

# Screenshots or Videos
https://user-images.githubusercontent.com/22690341/167119272-4a97613d-2675-466f-909f-dac7511a947c.mov

# Platform
OS: [e.g. macOS, Windows, Linux]
Browser: [e.g. Chrome, Safari, Firefox]
Version: [e.g. 10.2.0]

# Additional context

**Comment by bcakmakoglu** [comment]
I would guess that the moment you use fitView in your code the nodes haven't actually been updated yet so fitView calculates a boundary that's different to what you expect it to be (unlike hitting the button which will occur after updates).

You could try putting a `setTimeout(() => reactFlowInstance.fitView())` if that would work just to see if it's actually a timing issue 😅

Though I'd think there would be a more elegant solution to this.

**Reply by saomartinho** [reply]
I've tried it, even with 1000ms and it still doesn't work :/ 
I don't know what I'm doing wrong.

**Reply by moklick** [reply]
@saomartinho Can you create a codesandbox for this?

**Reply by saomartinho** [reply]
I will do it today.

**Reply by saomartinho** [reply]
the code is exactly the same: in the codesandbox FitView does something... however does not appear connections between the children and node 2 in codesandbox..

 I don't understand...

https://codesandbox.io/s/react-flow-example-79fpmn?file=/src/components/Graph.logic.ts

**Reply by saomartinho** [reply]
So, I realized the reason for centering on codesandbox, basically locally had this:

```js
const toggleExpand = async (id: string) => {
    ...
    const nodesTest = nodesFlow.map((node) => {
    ...
}

...
 const onInit = async (reactFlowInstance: ReactFlowInstance) => {
    const els = await createGraphLayoutElk(reactFlowInstance.getNodes(), reactFlowInstance.getEdges());

    setNodes(els);
    reactFlowInstance.fitView();
};
```

and in codesanbox it looks like this:

```js
const toggleExpand = async (id: string) => {
    ...
    const nodesTest = reactFlowInstance.getNodes().map((node) => {
    ...
}

...
 const onInit = async (reactFlowInstance: ReactFlowInstance) => {
    const els = await createGraphLayoutElk(initialNodes, initialEdges);

    setNodes(els);
    reactFlowInstance.fitView();
};
```

But I still couldn't figure out the motive... I replicated and started centering when it expands. However now, when I collapse the node, the layout is strange and when I expand again, the layout of the children is also strange... (you can see the video).




https://user-images.githubusercontent.com/22690341/167863820-26183a96-8326-48c0-875d-dfc7f15659e2.mp4


## Update
I noticed now, that the layouts I pass locally to elkjs, such as spacing, paddings, look different compared to codesandbox. And I still couldn't figure out the reason...
